Remove duplicate secret usages in CI build

### Description

Sounds like build is broken on `main`: https://github.com/clementguillot/nx-cloud-ce/actions/runs/8016366697/job/21898699961
Error is
```log
Error: [@octokit/auth-action] The token variable is specified more than once. Use either `with.token`, `with.GITHUB_TOKEN`, or `env.GITHUB_TOKEN`. See https://github.com/octokit/auth-action.js#createactionauth
Error: Process completed with exit code 1.
```

We should refactor CI build configuration to avoid duplicated usages of `${{ secrets.GITHUB_TOKEN }}`.

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -38,6 +38,8 @@ jobs:
       - build
     if: github.event_name == 'pull_request'
     runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     steps:
       - uses: actions/checkout@v4
         with:
@@ -49,12 +51,12 @@ jobs:
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
+          password: ${{ env.GITHUB_TOKEN }}
       - uses: ./.github/actions/nx-affected-target
         with:
           target: container
         env:
-          INPUT_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          INPUT_GITHUB_TOKEN: ${{ env.GITHUB_TOKEN }}
 
   test:
     runs-on: ubuntu-latest
@@ -91,6 +93,8 @@ jobs:
     permissions:
       contents: read
       packages: write
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     steps:
       - uses: actions/checkout@v4
         with:
@@ -102,10 +106,10 @@ jobs:
         with:
           registry: ghcr.io
           username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
+          password: ${{ env.GITHUB_TOKEN }}
       - uses: ./.github/actions/nx-affected-target
         with:
           target: container
         env:
-          INPUT_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          INPUT_GITHUB_TOKEN: ${{ env.GITHUB_TOKEN }}
           INPUT_PUSH: true

